sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+* Implement cluster level Pod Security to prevent host mount access to pod
+  * Status: implemented / planned / to be implemented / End User Guidance
+
+#### STRIDE-INFODISCLOSE-3 - Exposure of Management and Workload Cluster CA private keys and certs
+
+Exposure of Management Cluster CA private keys and certs on the node or in the
+cloud resources like Secrets Manager can lead to full or partial control over
+one or more clusters.
+
+#### STRIDE-INFODISCLOSE-3 Recommended Mitigations
+
+* Keep CA key/cert pairs short lived or deleted after use from Secrets Manager
+* Shorten CA cert lifetime
+* Disable or restrict SSH access to nodes in a cluster
+* Audit access or edits to CA private keys and cert files located on the node
+* Implement cluster level Pod Security to prevent host mount access to pod
+  * Status: implemented / planned / to be implemented / End User Guidance
+
+#### STRIDE-INFODISCLOSE-4 - Exposure of Cloud credentials that allow privileged actions
+
+Cloud credentials consumed by cluster API components can be leaked or
+exfiltrated to spoof Cluster API components or launch a denial of service attack
+
+#### STRIDE-INFODISCLOSE-4 Recommended Mitigations
+
+* Use short-lived credentials that are auto-renewed using node level attestation
+  * Status: implemented / planned / to be implemented / End User Guidance
+* Disable or restrict SSH access to nodes in a cluster
+  * Status: implemented / planned / to be implemented / End User Guidance
+* Implement cluster level Pod Security to prevent host mount access to pod
+  * Status: implemented / planned / to be implemented / End User Guidance
+
+### Denial of Service
+
+#### STRIDE-DOS-1 - Misuse of cloud credentials to create unlimited number of cloud resources
+
+Cloud credentials can be misused by someone who has access to it by accidentally
+or maliciously creating very large number of cloud resources to exhaust cloud
+capacity or for financial damage caused due to surprise bills
+
+#### STRIDE-DOS-1 Recommended Mitigations
+
+* Create safe defaults (soft and hard) for the maximum number of cloud resources
+  that can be created.
+* Apply rate limits on how many “create cloud resources” requests can be
+  successfully processed using the same cloud credentials.
+* Implement threshold based alerts when a particular soft limit is exceeded
+  * Status: implemented / planned / to be implemented / End User Guidance
+
+#### STRIDE-DOS-2 - Misuse of Infra Admin credentials to create unlimited number of clusters
+
+Infra Admin credentials can be misused by someone who has access to it by
+accidentally or maliciously creating very large number of clusters to exhaust
+capacity or slow down the operations team or for financial damage caused due to
+surprise bills
+
+#### STRIDE-DOS-2 Recommended Mitigations
+
+* Create safe defaults (soft and hard) for the maximum number of clusters that
+  can be created using a management cluster
```

Comment:
If the number of clusters is limited they can just scale up a small number of clusters to a very large machine/node count